src/app/components/Menu/Menu.tsx

```diff
@@ -9,6 +9,34 @@ const Menu = () => {
   const [isOpen, setIsOpen] = useState(false);
   const [showLanguages, setShowLanguages] = useState(false);
   const route = useRouter();
+  const headerRef = useRef<HTMLDivElement>(null);
+  const headerMobileRef = useRef<HTMLDivElement>(null);
+  const prevScrollpos = useRef<number>(window.scrollY);
+
+  const handleScroll = useCallback(() => {
```

Comment:
Co ten useCallback dokładnie robi i jak to działa bez niego?

Reply:
callback zapisuje mi tutaj tą funkcję między renderami, nie potrzebuję depsów bo nie zmienia się logika